include/nbl/video/IGPUCommandPool.h

```diff
@@ -131,34 +147,47 @@ class NBL_API IGPUCommandPool : public core::IReferenceCounted, public IBackendO
             return cmdMem;
         }
 
-        inline void setNext(CCommandSegment* segment) { header.m_next = segment; }
-        inline CCommandSegment* getNext() const { return header.m_next; }
-        inline ICommand* getFirstCommand() { return reinterpret_cast<ICommand*>(m_data); } // TODO(achal): Probably remove?
-        inline uint8_t* getData() { return m_data; }
+        inline void setNext(CCommandSegment* segment) { m_header.next = segment; }
+        inline CCommandSegment* getNext() const { return m_header.next; }
+
+        inline void setNextHead(CCommandSegment* segment) { m_header.nextHead = segment; }
+        inline CCommandSegment* getNextHead() const { return m_header.nextHead; }
+
+        inline void setPrevHead(CCommandSegment* segment) { m_header.prevHead = segment; }
+        inline CCommandSegment* getPrevHead() const { return m_header.prevHead; }
```

Comment:
its best if you have a 
```cpp
static void link(CCommandSegment* prev, CCommandSegment* next)
{
   prev->nextHead = next;
   next->prevHead = prev;
}
```
which does everything at once, it should handle nullptrs gracefully since no refcounting.

Then in the pool you can do
```cpp
if (firstHead==head)
   firstHead = head->getNextHead();
CCommandSegment::link(head->getPrevHead(),head->getNextHead());
// delete impl of head
```
when deleting, and
```cpp
CCommandSegment::link(newHead,firstHead);
firstHead = newHead;
```

Reply:
also instead of `setNext` make the constructor take a `CCommandSegment* prev` and do
```cpp
prev->m_header.next = this
```

Reply:
added the `if` check clauses for the pointers

Reply:
Resolved.